### PR TITLE
Issue #4 updated codes to make python3 compatibility.

### PR DESCRIPTION
**Pull request by nazrulworld**
* flake8 cleanup [nazrulworld]

### CODE REVIEW

**Comment by jamadden** [comment]
Thanks for the PR! It will need a few more things to make it complete:

- Please add the versions of Python 3 to .travis.yml and make sure the tests pass there ( [zope.schema is a good example](https://github.com/zopefoundation/zope.schema/blob/master/.travis.yml) )
- Likewise for tox.ini
- The classifiers in setup.py need to be updated as well ([example](https://github.com/zopefoundation/zope.schema/blob/master/setup.py#L63))

**Comment by nazrulworld** [comment]
Thanks for your valuable feedback! I will do accordingly

**Comment by nazrulworld** [comment]
@jamadden added python2/3 compatibility codes added.

File: src/z3c/saconfig/zcml.py
Comment (ale-rt) [inline src/z3c/saconfig/zcml.py]: I think this is not necessary: u"Database URL" is equal to six.text_type("Database URL") in both Python versions

File: buildout.cfg
Comment (ale-rt) [inline buildout.cfg]: I see you are commenting `#    pysqlite` here but then a version pin is added below. Looks strange :)

File: src/z3c/saconfig/README.rst
Comment (ale-rt) [inline src/z3c/saconfig/README.rst]: Probably you should use from io import BytesIO and be sure that you are passing bytes to it.
See this patch as an example https://github.com/plone/plone.supermodel/commit/1d02b25d02b825cd64fa95760b0abee1b5055e7e#diff-bc1d78c2a92695a5b3d8a63f5ea6ab0fR16

**Review by ale-rt — CHANGES_REQUESTED** [review]
Thanks a lot :)
It looks really well done to me, there are just a few things to tune up.

In addition be sure you have signed the Zope contributor agreemeent: https://docs.zope.org/developer/becoming-a-committer.html

File: src/z3c/saconfig/zcml.py
Comment (nazrulworld) [reply src/z3c/saconfig/zcml.py]: You are right in Python 3.3+ it's legal again to use u''. I will revert that

**Comment by nazrulworld** [comment]
@ale-rt I have mailed with Signed copy Zope Committer Agreement

File: .travis.yml
Comment (mgedmin) [inline .travis.yml:6]: You can add 3.7 if you do the magic dance of

```
matrix:
    include:
        - python: 3.7
          dist: xenial
          sudo: required
```

Comment (mgedmin) [reply .travis.yml:6]: (It's fine if you don't want to do that at this time.)

File: CHANGES.rst
Comment (mgedmin) [inline CHANGES.rst]: Would be nice to mention which specific Python 3 versions are supported (AFAICT they're 3.5 and 3.6).

Usually we phrase this as

> - Added Python 3.5 and 3.6 compatibility

File: setup.py
Comment (mgedmin) [inline setup.py]: IMHO a GitHub link would be better -- this is the URL shown on PyPI, and it's a little pointless to have a PyPI page point back to itself.

File: src/z3c/saconfig/README.rst
Comment (mgedmin) [inline src/z3c/saconfig/README.rst]: No need for `six.b`, `b"..."` literals work across all currently supported Python versions.

File: src/z3c/saconfig/README.rst
Comment (mgedmin) [inline src/z3c/saconfig/README.rst]: There's a way of using `from __future__ import print_function` in doctests, then you could use `print()`.

(Or if you refrain from using multiple args, you can always do `print("args: {}".format(event.engine_args))` etc. and it'll work on both major Python versions without future imports.)

File: src/z3c/saconfig/tests.py
Comment (mgedmin) [inline src/z3c/saconfig/tests.py:22]: This seems wrong: PEP-8 suggests putting standard library imports (doctest, os, unittest) in a group above 3rd-party library imports (zope.*).

File: src/z3c/saconfig/utility.py
Comment (mgedmin) [inline src/z3c/saconfig/utility.py:15]: (Likewise)

File: src/z3c/saconfig/zcml.py
Comment (mgedmin) [inline src/z3c/saconfig/zcml.py:37]: Why the `+`s?  String concatenation is a perfectly cromulent syntactic feature.

File: src/z3c/saconfig/zcml.py
Comment (mgedmin) [inline src/z3c/saconfig/zcml.py]: I don't like this.

It would be better to

```
if isinstance(setup, bytes):
    setup = setup.decode()
```

**Review by mgedmin — APPROVED** [review]
LGTM, with minor possible improvements.

File: src/z3c/saconfig/README.rst
Comment (mgedmin) [reply src/z3c/saconfig/README.rst]: (There are two ways of using print_function in doctests: you can just write it out in the doctest itself like [zc.relation does](https://github.com/zopefoundation/zc.relation/blob/master/src/zc/relation/README.rst), or you can stuff it into globs like [zope.mimetype does](https://github.com/zopefoundation/zope.mimetype/blob/master/src/zope/mimetype/tests/test_doctests.py#L60).)

File: src/z3c/saconfig/tests.py
Comment (mgedmin) [inline src/z3c/saconfig/tests.py]: This is the only remaining place where you use `six`, and it can be rewritten to use `print()` directly.

Comment (nazrulworld) [reply src/z3c/saconfig/tests.py]: thanks a lot :) now six dependency is wiped

**Review by mgedmin — APPROVED** [review]
Feel free to merge when your commit bit comes through!

**Review by ale-rt — APPROVED** [review]

**Comment by nazrulworld** [comment]
@ale-rt I am now approved as member as committer. Can you please initiate to merge this PR. Thanks

**Comment by jamadden** [comment]
The general policy in zopefoundation repositories is that once the PR is approved, the person that opened the PR clicks the "merge" button. This serves to prove that they've signed the committer agreement and have the correct permissions. (Put another way, we usually merge our own PRs after they've been approved by others.) Feel free to click the merge button!

**Pull request merged**
